src/cryptonote_basic/miner.cpp

```diff
@@ -524,7 +523,7 @@ namespace cryptonote
   //-----------------------------------------------------------------------------------------------------
   bool miner::worker_thread()
   {
-    uint32_t th_local_index = boost::interprocess::ipcdetail::atomic_inc32(&m_thread_index);
+    uint32_t th_local_index = m_thread_index++;
```

Comment:
```suggestion
    const uint32_t th_local_index = m_thread_index;
    ++m_thread_index;
```

Reply:
I thought about making the value `const` but it didn't match the rest of the code. Should I go ahead and do it anyways?

Reply:
If you're touching the line, might as well go all the way.

Reply:
Same. It must be done in a single operation, not two lines of code!